lib/cn/cn_metrics.h

```diff
@@ -9,152 +9,173 @@
+    uint64_t ns_hclen;     //<! estimated total hblock capacity (mpr_alloc_cap) after compaction
+    uint64_t ns_kclen;     //<! estimated total kblock capacity (mpr_alloc_cap) after compaction
+    uint64_t ns_vclen;     //<! estimated total vblock capacity (mpr_alloc_cap) after compaction
+    uint16_t ns_pcap;      //<! current size / max size as a percentage (0 <= ns_pcap <= 100)
 };
 
 /**
- * Node derived stats:
- *   cn_ns_clen()  - estimated mpool capacity used by node after kv-compaction
- *   cn_ns_alen()  - current capacity used by node (sum of kvset alen)
- *   cn_ns_wlen()  - sum of kvset wlen
- *   cn_ns_vulen() - sum of kvset vblocks ulen
- *   cn_ns_keys()  - number of keys in kvset
- *   cn_ns_kblks() - number of kblocks in node
- *   cn_ns_vblks() - number of vblocks in node
+ * Sum of "alen" for all mblocks in node.
  */
-static inline u64
+static inline uint64_t
 cn_ns_alen(const struct cn_node_stats *ns)
 {
     return kvset_alen(&ns->ns_kst);
 }
 
-static inline u64
+/**
+ * Sum of "wlen" for all mblocks in node.
+ */
+static inline uint64_t
 cn_ns_wlen(const struct cn_node_stats *ns)
 {
     return kvset_wlen(&ns->ns_kst);
 }
 
-static inline u64
+/**
+ * Total vlbock data in node referenced by keys in the node.
+ */
+static inline uint64_t
 cn_ns_vulen(const struct cn_node_stats *ns)
 {
     return kvset_vulen(&ns->ns_kst);
 }
 
-static inline u64
+/**
+ * Estimated mpool capacity used by node after kv-compaction.
+ */
+static inline uint64_t
 cn_ns_clen(const struct cn_node_stats *ns)
 {
-    /* This is different than alen, wlen, and ulen because it depends
-     * on HyperLogLog stats which are only computed at the node level.
-     */
     return ns->ns_hclen + ns->ns_kclen + ns->ns_vclen;
 }
 
-static inline u64
+/**
+ * Number of keys in node (counts duplicates).
+ */
+static inline uint64_t
 cn_ns_keys(const struct cn_node_stats *ns)
 {
     return ns->ns_kst.kst_keys;
 }
 
+/**
+ * An estimate of the number of unique keys in node.
+ *
+ * Not accurate unless HyperLogLog stats are enabled on the node (for example,
+ * on the root node).
+ */
 static inline uint64_t
 cn_ns_keys_uniq(const struct cn_node_stats *ns)
 {
     return ns->ns_keys_uniq;
 }
 
+/**
+ * Number of tombstones in node.
+ */
 static inline uint64_t
 cn_ns_tombs(const struct cn_node_stats *ns)
 {
     return ns->ns_kst.kst_tombs;
 }
 
+/**
+ * Number of hblocks in node.
+ */
 static inline uint32_t
 cn_ns_hblks(const struct cn_node_stats *ns)
 {
     return ns->ns_kst.kst_hblks;
 }
 
+/**
+ * Number of kblocks in node.
+ */
 static inline uint32_t
 cn_ns_kblks(const struct cn_node_stats *ns)
 {
     return ns->ns_kst.kst_kblks;
 }
 
+/**
+ * Number of vblocks in node.
+ */
 static inline uint32_t
 cn_ns_vblks(const struct cn_node_stats *ns)
 {
     return ns->ns_kst.kst_vblks;
 }
 
+/**
+ */
+static inline uint32_t
+cn_ns_kvsets(const struct cn_node_stats *ns)
```

Comment:
It would be a good idea to review all callers to make sure they are using the right type.

Reply:
I started down that path, but it quickly escalated so I'm saving it for another day.  If callers were okay with a 64-bit value, they'll be fine with 32-bit value.

Reply:
Macro comment, but we have these types of mismatches all over the codebase. Lots of fixing to do in this area.